### PR TITLE
[Mac] Input validation updates for TextFields

### PR DESCRIPTION
**Pull request by blue-stone99**
## Related Issue
Please use one of the well-known [github fixes keywords](https://help.github.com/en/articles/closing-issues-using-keywords) to reference
the issues fixed with this PR (eg Fixes #<github issue number>). If an issue doesn't yet exist please create one if reasonable to aid 
in issue tracking.

**NOTE**: For multiple issues resolved by this PR use the corresponding keywords **every time** in a comma-delimited list per the reference
page above.

## Description
Create a [descriptive title and provide all of the relevant details in the description](https://chris.beams.io/posts/git-commit/). 
This information lives with the code and will help future readers (***including yourself***) and will serve as documentation.

At the very least please describe the issue you are addressing, and what your fix entails. 

image
image
image
image
image

## Sample Card
If appropriate, please include a link to a card in one of the samples directories that can be used to validate this change. This can be an existing card or a card added with this PR.

## How Verified
How you verified the fix, including one or all of the following: 
1. New unit tests that were added if any. If none were added please add a quick line explaining why not.
2. Existing relevant unit/regression tests that you ran
3. Manual scenario verification if any; ***Do include .gif's or screenshots of the testing you performed here if you think that it 
will aid in code reviews or corresponding fixes on other platforms for eg.***

### CODE REVIEW

**Comment by blue-stone99** [comment]
Error text taken from RenderConfig is handled by @mukuagar in #246

File: source/macos/AdaptiveCards/AdaptiveCards/Renderers/InputNumberRenderer.swift [hunk 1]
```diff
@@ -202,22 +228,13 @@ extension ACRNumericTextField: InputHandlingViewProtocol {
         stepper.valueWraps = shouldWrap
     }
     
-    func setId(idString: String?) {
-        self.id = idString
-    }
```
Comment (mukuagar) [inline source/macos/AdaptiveCards/AdaptiveCards/Renderers/InputNumberRenderer.swift:207]: Never would have noticed it's a ghost function🔥

File: source/macos/AdaptiveCards/AdaptiveCards/Renderers/TextInputRenderer.swift [hunk 2]
```diff
@@ -155,14 +162,24 @@ class ACRTextInputView: ACRTextField, InputHandlingViewProtocol {
...
     var idString: String?
+    var regex: String?
     
     override func textDidChange(_ notification: Notification) {
         super.textDidChange(notification)
+        if isValid {
+            ACRView.focusedElementOnHideError = self
+            errorMessageHandler?.hideErrorMessage(for: self)
+            hideError()
+        }
```
Comment (mukuagar) [inline source/macos/AdaptiveCards/AdaptiveCards/Renderers/TextInputRenderer.swift:181]: This is very very clean

File: source/macos/AdaptiveCards/AdaptiveCards/Views/ACRDateField.swift [hunk 3]
```diff
@@ -112,18 +117,18 @@ class ACRDateField: NSView, InputHandlingViewProtocol {
     }
     
     var isValid: Bool {
-        return true
+        // TODO: Add min and max date checks too
+        return isBasicValidationsSatisfied
```
Comment (mukuagar) [inline source/macos/AdaptiveCards/AdaptiveCards/Views/ACRDateField.swift:121]: Should we add min and max check here and move it away from UI?

**Review by mukuagar — COMMENTED** [review]
Overall looks good to me

File: source/macos/AdaptiveCards/AdaptiveCards/Views/ACRDateField.swift (hunk 3 above)
Comment (blue-stone99) [reply source/macos/AdaptiveCards/AdaptiveCards/Views/ACRDateField.swift:120]: Nope not really. Just as a extra safety net.
Ideally that condition would never be met.

**Review by mukuagar — APPROVED** [review]

**Pull request merged**